.changeset/rare-ducks-sip.md

```diff
@@ -0,0 +1,49 @@
+---
+'astro': patch
+---
+
+Adds a new Astro Adapter Feature called `experimentalStaticHeaders` to allow your adapter to control serving CSP in static pages.
+
+For adapters that enable support for this feature, Astro will not serve the CSP `<meta http-equiv="content-security-policy">` element in static pages.
+
+Instead, Astro will serve the value of the header inside a map that can be retrieved from the hook `astro:build:generated`. Adapters can read this mapping and use their hosting headers capabilities to create a configuration file.
+
+The name of the new field is called `experimentalRouteToHeaders` and it contains a map of `Map<IntegrationResolvedRoute, Headers>` where
+the `Headers` contain the headers emitted during the rendering of the static pages.
+
+A new field called `experimentalRouteToHeaders` will contain a map of `Map<IntegrationResolvedRoute, Headers>` where the `Headers` type contains the headers emitted by the rendered static route. 
```

Comment:
```suggestion
A new field called `experimentalRouteToHeaders` will contain a map of `Map<IntegrationResolvedRoute, Headers>` where the `Headers` type contains the headers emitted by the rendered static route. 
```
Looks like an old and new version of a sentence here. If correct, the second one is the better sentence!